src/main/java/com/systemmeltdown/robotlog/outputs/ZipFileUtils.java

```diff
@@ -0,0 +1,163 @@
+
+      if (parentPath == null) {
+        System.err.println("ZipFileOutput: Cannot create dir (" + path + ") for unknown parent dir");
+        return null;
+      }
+
+      File parentDir = getOrCreateDir(parentPath);
+      if (parentDir == null) {
+        // No need to log to stderr here, the parent recursive call should have already logged.
+        return null;
+      }
+
+      System.out.println("ZipFileOutput: Creating directory '" + path + "'");
+      if (!dir.mkdir()) {
+        System.err.println("ZipFileOutput: Failed to create parent directory: " + parentPath);
+        return null;
+      }
+    }
+
+    if (!dir.isDirectory()) {
+      System.err.println("ZipFileOutput: Path is not directory: " + path);
+      return null;
+    }
+    if (!dir.canWrite() || !dir.canExecute()) {
+      System.err.println("ZipFileOutput: Insufficient privileges for directory: " + path);
+      return null;
+    }
+
+    // Everything checks out.
+    return dir;
+  }
+
+  public static String printHeader(Map<String, Object> header) {
+    StringBuffer sb = new StringBuffer();
+    boolean needsComma = false;
+
+    sb.append("\"header\": { ");
+
+    for (Map.Entry<String, Object> entry : header.entrySet()) {
+      String keyStr = "\"" + entry.getKey() + "\"";
+      String valueStr = "\"" + printValue(entry.getValue(), entry.getValue().getClass()) + "\"";
+
+      if (needsComma) {
+        sb.append(", ");
+      }
+      sb.append(keyStr + ": " + valueStr);
+      needsComma = true;
+    }
+
+    sb.append(" }");
+    return sb.toString();
+  }
+
+  public static String printTopic(String topicName, Class<?> valueType) {
+    String topicStr = "\"" + topicName + "\"";
+    String valueTypeStr = "\"" + printValueType(valueType) + "\"";
+
+    return "{ \"name\":" + topicStr + ", \"type\":" + valueTypeStr + " }";
+  }
+
+  public static String printEntry(String topicName, Object value, Class<?> valueType, double relativeNanos, double timeRoundingFactor) {
+    String topicStr = "\"" + topicName + "\"";
+    String valueStr = printValue(value, valueType);
+    double timeSeconds = Utils.roundByFactor(((double) relativeNanos) / 1000000000, timeRoundingFactor);
+    String timeStr = Double.toString(timeSeconds);
+
+    return "{ \"topic\":" + topicStr + ", \"value\":" + valueStr + ", \"time\":" + timeStr + " }";
+  }
+
+  public static String printValueType(Class<?> valueType) {
+    if (valueType == String.class) {
+      return "string";
+    }
+    if (valueType == Double.class) {
+      return "decimal";
+    }
+    if (valueType == Integer.class) {
+      return "integer";
+    }
+    if (valueType == Boolean.class) {
+      return "boolean";
+    }
+    System.err.println("ZipFileUtils.parseValueType: Unrecognized value type: " + valueType.getName());
+    return "";
+  }
+
+  public static String printValue(Object value, Class<?> valueType) {
+    if (valueType == String.class) {
+      return (String) value;
```

Comment:
Do we need to worry about escaping quotes in value?

Reply:
Good catch, thanks!